### ISSUE TITLE
about the loss in the second stage of training

### ISSUE THREAD

**Issue by jun000697** [indu1ge/DepthMaster#15]
I followed readme to reproduce the project and when I do stage 2 training, the loss contains nan.

<img width="1155" height="235" alt="Image" src="https://github.com/user-attachments/assets/270d6e47-10c6-430e-9a13-27d761eb0b79" />
so,What should I do to resume normal training?

**Comment by jun000697** [comment]
What does the warning in this image mean? Does it mean that I didn't use a custom UNet (including blockfe) and instead trained using the officially defined UNet, or can this warning be ignored?

**Comment by indu1ge** [comment]
Regarding the abnormal loss issue, could you please provide the first-stage model you trained?
As for the warning about the U-Net weights loading, you can ignore it. Since the first-stage model does not contain the blockfe module, those weights will be newly initialized.

**Comment by jun000697** [comment]
I did not undergo the first stage of training and went directly to the second stage, so I do not have the weights from the first stage.




双控研2501班郭君君
***@***.***

**Comment by indu1ge** [comment]
Hi,

Regarding the nan loss issue, you can try commenting out Lines 325 and 326 in `train_s2.py`.

Additionally, we have fixed several other issues. Please refer to my recent commit: 63999f63553bbab4ca80207c9b47534253abfae3 (click the hash to view the changes).

After updating the code, the training should be normal, and the loss should be around 0.006.

Hope this helps!

**Comment by jun000697** [comment]
Thank you for your reply. I also noticed that the depth relationship in your images seems to be represented differently in colors. Why is the depth near blue and the depth far red? Also, my training speed seems a bit slow. My device is an RTX 4090 graphics card. Could you tell me how to improve the training speed?




双控研2501班郭君君
***@***.***

**Comment by indu1ge** [comment]
Hi,

We choose different colormaps to better highlight near details. You can change the colormap in the visualization code.

For the slow training speed, maybe you could try :
 - Enabling mixed precision
 - Change a device with larger VRAM so that you can increase batch size
 - Reduce the number of training iterations.

**Comment by jun000697** [comment]
Hi, even after I commented out those two lines of code as you suggested, the loss = Nan still occurred.




双控研2501班郭君君
***@***.***

**Comment by indu1ge** [comment]
It's suggested to undergo the first-stage training and use the first-stage model to initialize the model weights.

**Comment by jun000697** [comment]
Why?




双控研2501班郭君君
***@***.***

**Comment by indu1ge** [comment]
The current setting and training code is suited for our two-stage training with the first stage learning basic structure and the second stage refining details. If you directly train the second-stage model, maybe you should modify training settings accordingly.

**Comment by jun000697** [comment]
Hello, when I ran the first stage of training, I also encountered a loss of NaN. How can I resolve this?




双控研2501班郭君君
***@***.***

**Comment by indu1ge** [comment]
We have tested the training code, and the loss output is normal. Maybe you could check the data, weights and environment setup.
